File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        
        map = {}

        for ele in nums:
            if ele in map:
                map[ele] += 1
            else:
                map[ele] = 1
        
        ans = []
        for i in range(k):
            maxele = -1
            fillele = -1
            for key in map.keys():
                if map[key] > maxele:
                    maxele = map[key]
                    fillele = key
            ans.append(fillele)
            map.pop(fillele)
        return ans

File: 347-top-k-frequent-elements/top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] ans = new int[k];

        for (int ele : nums) {
            if (map.containsKey(ele)) {
                int freq = map.get(ele);
                map.put(ele, freq + 1);
            } else {
                map.put(ele, 1);
            }
        }
        for (int i = 0; i < k; i++) {
            int maxfreq = -1;
            int fill = -1;
            for (int key : map.keySet()) {
                if (map.get(key) >= maxfreq) {
                    maxfreq = map.get(key);
                    fill = key;
                }
            }
            ans[i] = fill;
            map.remove(fill);
        }
        return ans;
    }
}